aws/docker.ts

```diff
@@ -303,23 +311,35 @@ interface CommandResult {
 async function runCLICommand(
     cmd: string,
     args: string[],
-    returnStdout?: boolean,
+    resource: pulumi.Resource,
     stdin?: string): Promise<CommandResult> {
+
+    // Generate a unique stream-ID that we'll associate all the docker output with. This will allow
+    // each spawned CLI command's output to associated with 'resource' and also streamed to the UI
+    // in pieces so that it can be displayed live.  The stream-ID is so that the UI knows these
+    // messages are all related and should be considered as one large message (just one that was
+    // sent over in chunks).
+    //
+    // We use Math.random here in case our package is loaded multiple times in memory (i.e. because
+    // different downstream dependencies depend on different versions of us).  By being random we
+    // effectively make it completely unlikely that any two cli outputs could map to the same stream
+    // id.
+    const streamID = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
```

Comment:
Looks like the RPC interface uses `int32`: https://github.com/pulumi/pulumi/blob/66c91d610737b3a07570b763e9391dd934f207ca/pkg/diag/diag.go#L33

So using 53-bit integer here likely won't work right?  (not sure how gRPC JS client marshalls overflowing JS numbers to int32,?)

Reply:
ah, true.  i'll make this safer.